cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/queue/hbase/HBaseQueueAdmin.java

```diff
@@ -482,60 +479,51 @@ public void configureGroups(QueueName queueName, Map<Long, Integer> groupInfo) t
 
   @Override
   public void upgrade() throws Exception {
-    // For each table managed by this admin, performs an upgrade
-    for (HTableDescriptor desc : getHBaseAdmin().listTables()) {
-      String tableName = Bytes.toString(desc.getName());
+    // For each table managed by this admin, perform an upgrade
+    List<TableId> tableIds = tableUtil.listTables(getHBaseAdmin());
+    for (TableId tableId : tableIds) {
       // It's important to skip config table enabled.
-      if (isDataTable(tableName)) {
-        LOG.info(String.format("Upgrading queue hbase table: %s", tableName));
+      if (isDataTable(tableId)) {
+        LOG.info(String.format("Upgrading queue table: %s", tableId));
         Properties properties = new Properties();
+        HTableDescriptor desc = tableUtil.getHTableDescriptor(getHBaseAdmin(), tableId);
         if (desc.getValue(HBaseQueueAdmin.PROPERTY_PREFIX_BYTES) == null) {
           // It's the old queue table. Set the property prefix bytes to SALT_BYTES
           properties.setProperty(HBaseQueueAdmin.PROPERTY_PREFIX_BYTES, Integer.toString(HBaseQueueAdmin.SALT_BYTES));
         }
-        upgrade(tableName, properties);
-        LOG.info(String.format("Upgraded queue hbase table: %s", tableName));
+        upgrade(tableId, properties);
+        LOG.info(String.format("Upgraded queue table: %s", tableId));
       }
     }
   }
 
+  private void upgrade(TableId tableId, Properties properties) throws Exception {
```

Comment:
I just figured it'd be good to separate the logic of determining which tables need upgrading and then the actual table upgrade (even though its only 5 lines).